### DISCUSSION TITLE
New build system

### DISCUSSION THREAD

**Discussion by nordmoen** [NorESMhub/BLOM#165] [Announcements]
This issue should be seen as a meta issue for the discussion around a new build system for BLOM.

## Possible candidates
- [`CMake`](https://cmake.org/)
- [`SCons`](https://scons.org/)
- [`waf`](https://waf.io/)
- [`Meson`](https://mesonbuild.com/)
- `Make`

What we should discover is what are the requirements for BLOM's build system and discuss in this issue. The following is a small list compiled after a discussion with @matsbn. Feel free to add to this top post with new requirements.

## Requirements
- [ ] Must support Linux/Unix and Sigma2 resources, but should also build on local machines with Linux/macOS
- [ ] Must handle dependencies (e.g. `netCDF` or `MPI`)
- [ ] Should be able to automatically generate runtime files such as `dimensions.F`
- [ ] Should support externally updated files for configuration

**Comment by matsbn** [comment]
@nordmoen, I pushed a modified blom_dimensions script to branch [meson](https://github.com/NorESMhub/BLOM/tree/meson). It works as before on Sigma2 machines and macOS. Great if you can test it on Ubuntu and Fedora.

**Comment by nordmoen** [comment]
Seems to work on my Fedora machine, we can ensure that it works on Ubuntu with the CI setup in PR #64.
